src/base/net/downloadmanager.cpp

```diff
@@ -271,9 +274,12 @@ void Net::DownloadManager::handleDownloadFinished(DownloadHandlerImpl *finishedH
     }
 
     auto *handler = waitingJobsIter.value().dequeue();
-    qDebug("Downloading %s...", qUtf8Printable(handler->url()));
-    processRequest(handler);
     handler->disconnect(this);
+    QTimer::singleShot(m_sequentialServices.value(id, 0s) , this, [this, handler]
+    {
+        qDebug("Downloading %s...", qUtf8Printable(handler->url()));
+        processRequest(handler);
+    });
```

Comment:
Speculative question, should `disconnect()` happens after `processRequest()` or it doesn't matter?
```suggestion
    QTimer::singleShot(m_sequentialServices.value(id, 0s) , this, [this, handler]
    {
        qDebug("Downloading %s...", qUtf8Printable(handler->url()));
        processRequest(handler);
        handler->disconnect(this);
    });
```

Reply:
It is connected to be removed from waiting jobs queue in case it is destroyed before finished (e.g. when cancelled by user side). So it should be disconnected once we explicitly removed it from queue. Seems it is OK to disconnect after processRequest. The main thing is to do it before handler is deleted.